### PR TITLE
1.27 Cherry pick of #120492: service controller: improve node lifecycle updates - update

### PR DESCRIPTION
**Pull request by code-elinka**
Cherry pick of https://github.com/kubernetes/kubernetes/pull/120492 on release-1.27.

https://github.com/kubernetes/kubernetes/pull/120492: service controller: improve node lifecycle updates - update

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note
Service Controller: update load balancer hosts after node's ProviderID is updated
```

### CODE REVIEW

**Comment by thockin** [comment]
Thanks!

/lgtm
/approve

**Comment by thockin** [comment]
Needs relnote and cherry pick approval

**Comment by aojea** [comment]
/kind bug
/release-note none
ping @kubernetes/release-managers

**Comment by thockin** [comment]
@kubernetes/release-managers

**Comment by saschagrunert** [comment]
Let's fix the test failures in https://github.com/kubernetes/kubernetes/pull/120945 before approving this one.

**Comment by code-elinka** [comment]
> Let's fix the test failures in #120945 before approving this one.

I created https://github.com/kubernetes/kubernetes/pull/121483 instead because Cezary (the author ) is out

**Comment by thockin** [comment]
Thanks!

/lgtm
/approve

**Comment by saschagrunert** [comment]
Waiting for `/approve` by the SIG on https://github.com/kubernetes/kubernetes/pull/120943

**Comment by thockin** [comment]
#120492 was approved

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**
